test/python/tools/test_admin.py

```diff
@@ -70,3 +70,94 @@ def test_analyse_indexing_with_osm_id(project_env, temp_db_cursor):
+                      ((100, 'N', 'boundary', 'administrative'),
+                      (145, 'N', 'boundary', 'administrative'),
+                      (175, 'R', 'landcover', 'grass')))
+        table_factory('place_to_be_deleted',
+                      """osm_id BIGINT,
+                      osm_type CHAR(1),
+                      class TEXT NOT NULL,
+                      type TEXT NOT NULL,
+                      deferred BOOLEAN""")
+        temp_db_cursor.execute("""INSERT INTO placex (place_id, osm_id, osm_type, class, type, indexed_date, indexed_status)
+                              VALUES(1, 100, 'N', 'boundary', 'administrative', current_date - INTERVAL '1 month', 1),
+                               (2, 145, 'N', 'boundary', 'administrative', current_date - INTERVAL '1 month', 1),
+                               (3, 175, 'R', 'landcover', 'grass', current_date - INTERVAL '1 month', 1)""")
+        temp_db_cursor.execute("""CREATE OR REPLACE FUNCTION flush_deleted_places()
+                               RETURNS INTEGER
+                               AS $$
+                               BEGIN
+                                UPDATE placex p SET indexed_status = 100 FROM place_to_be_deleted d
+                                WHERE p.osm_type = d.osm_type
+                                AND p.osm_id = d.osm_id
+                                AND p.class = d.class
+                                AND p.type = d.type
+                                AND NOT deferred;
+                               TRUNCATE TABLE place_to_be_deleted;
+                                RETURN NULL;
+                               END;
+                               $$
+                               LANGUAGE plpgsql;
+                               """)
+        temp_db_cursor.execute("""CREATE OR REPLACE FUNCTION place_force_delete(placeid BIGINT)
+                               RETURNS BOOLEAN
+                               AS $$
+                               DECLARE 
+                                osmid BIGINT;
+                                osmtype character(1);
+                                pclass text;
+                                ptype text;
+                               BEGIN
+                                SELECT osm_type, osm_id, class, type FROM placex WHERE place_id = placeid INTO osmtype, osmid, pclass, ptype;
+                                DELETE FROM import_polygon_delete WHERE osm_type = osmtype AND osm_id = osmid AND class = pclass AND type = ptype;
+                                INSERT INTO place_to_be_deleted (osm_type, osm_id, class, type, deferred)
+                                        VALUES(osmtype, osmid, pclass, ptype, false);
+                                PERFORM flush_deleted_places();
+                                RETURN TRUE;
+                               END;
+                               $$
+                               LANGUAGE plpgsql;
+                               """)
```

Comment:
Can you try out if it is possible to use the original code here, so that we catch in the unit tests, when something breaks in the SQL? The following should work:

* move `flush_deleted_places()` from `lib-sql/functions/place_triggers.sql` in to `lib-sql/functions/utils.sql`
* load the original `utils.sql` here similar as it is done in  https://github.com/osm-search/Nominatim/blob/master/test/python/tokenizer/test_icu.py#L100

Reply:
I moved the function over and it seems to be working fine. I had to add a couple of extra tables and functions to the setup for the unit tests to cover everything in utils.sql.